File: 169 Majority Element/169_Majority_Element.py
class Solution:
    def majorityElement(self, nums):
        _count = {}
        for i in nums:
            if _count.get(i):
                _count[i] += 1
            else:
                _count[i] = 1
        return max(_count, key=_count.get)


# AFTER SUBMISSION 
# Runtime: 172 ms, faster than 52.14% of Python3 online submissions for Majority Element.
# Memory Usage: 15.4 MB, less than 82.70% of Python3 online submissions for Majority Element.

File: 169 Majority Element/169_Majority_Element.go
package main

func majorityElement(nums []int) int {
	m := make(map[int]int)
	for _, n := range nums {
		m[n]++
	}
	KEY := 0
	VALUE := 0
	for key, value := range m {
		if VALUE < value {
			KEY = key
			VALUE = value
		}
	}
	return KEY
}

func main() {
	majorityElement([]int{3, 2, 3})
}

// Runtime: 27 ms, faster than 18.52% of Go online submissions for Majority Element.
// Memory Usage: 6.4 MB, less than 14.60% of Go online submissions for Majority Element.
// Next challenges:
